src/transformers/models/qwen2/modeling_qwen2.py

```diff
@@ -329,6 +329,9 @@ def forward(self, x, position_ids):
             self._dynamic_frequency_update(position_ids, device=x.device)
 
         # Core RoPE block
+        if self.inv_freq.device != x.device:
+            self.inv_freq = self.inv_freq.to(x.device)
+
         inv_freq_expanded = self.inv_freq[None, :, None].float().expand(position_ids.shape[0], -1, 1)
```

Comment:
hmm.. I'm not sure about modifying the device for `self.inv_freq` in forward, instead I think it's safer to use 

```suggestion
        inv_freq = self.inv_freq.to(x.device)
        inv_freq_expanded = inv_freq[None, :, None].float().expand(position_ids.shape[0], -1, 1)
```